client/apis/volumesnapshot/v1/types.go

```diff
@@ -0,0 +1,429 @@
+	// snapshot controller with the "creation_time" value returned from CSI
+	// "CreateSnapshot" gRPC call.
+	// For a pre-existing snapshot, this field will be filled with the "creation_time"
+	// value returned from the CSI "ListSnapshots" gRPC call if the driver supports it.
+	// If not specified, it indicates that the creation time of the snapshot is unknown.
+	// +optional
+	CreationTime *metav1.Time `json:"creationTime,omitempty" protobuf:"bytes,2,opt,name=creationTime"`
+
+	// readyToUse indicates if the snapshot is ready to be used to restore a volume.
+	// In dynamic snapshot creation case, this field will be filled in by the
+	// snapshot controller with the "ready_to_use" value returned from CSI
+	// "CreateSnapshot" gRPC call.
+	// For a pre-existing snapshot, this field will be filled with the "ready_to_use"
+	// value returned from the CSI "ListSnapshots" gRPC call if the driver supports it,
+	// otherwise, this field will be set to "True".
+	// If not specified, it means the readiness of a snapshot is unknown.
+	// +optional
+	ReadyToUse *bool `json:"readyToUse,omitempty" protobuf:"varint,3,opt,name=readyToUse"`
+
+	// restoreSize represents the minimum size of volume required to create a volume
+	// from this snapshot.
+	// In dynamic snapshot creation case, this field will be filled in by the
+	// snapshot controller with the "size_bytes" value returned from CSI
+	// "CreateSnapshot" gRPC call.
+	// For a pre-existing snapshot, this field will be filled with the "size_bytes"
+	// value returned from the CSI "ListSnapshots" gRPC call if the driver supports it.
+	// When restoring a volume from this snapshot, the size of the volume MUST NOT
+	// be smaller than the restoreSize if it is specified, otherwise the restoration will fail.
+	// If not specified, it indicates that the size is unknown.
+	// +optional
+	RestoreSize *resource.Quantity `json:"restoreSize,omitempty" protobuf:"bytes,4,opt,name=restoreSize"`
+
+	// error is the last observed error during snapshot creation, if any.
+	// This field could be helpful to upper level controllers(i.e., application controller)
+	// to decide whether they should continue on waiting for the snapshot to be created
+	// based on the type of error reported.
+	// The snapshot controller will keep retrying when an error occurrs during the
+	// snapshot creation. Upon success, this error field will be cleared.
+	// +optional
+	Error *VolumeSnapshotError `json:"error,omitempty" protobuf:"bytes,5,opt,name=error,casttype=VolumeSnapshotError"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshotClass specifies parameters that a underlying storage system uses when
+// creating a volume snapshot. A specific VolumeSnapshotClass is used by specifying its
+// name in a VolumeSnapshot object.
+// VolumeSnapshotClasses are non-namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
```

Comment:
Currently, the storage and served version true is set to the v1beta1. v1 should have `served: true` and `storage: true` and v1beta1 should only have served: true. To do that
```suggestion
// +kubebuilder:resource:scope=Cluster
// +kubebuilder:storageversion
```